src/ribbon.c

```diff
@@ -19,86 +20,566 @@ typedef struct {
-    [RIBBON_SMART_GREAT] = { 0x88, 0x5A, 0x3, ribbon_name_smart_great, 0x77 },
-    [RIBBON_SMART_ULTRA] = { 0x89, 0x5B, 0x3, ribbon_name_smart_ultra, 0x78 },
-    [RIBBON_SMART_MASTER] = { 0x8A, 0x5C, 0x3, ribbon_name_smart_master, 0x79 },
-    [RIBBON_TOUGH] = { 0x8B, 0x59, 0x4, ribbon_name_tough, 0x7A },
-    [RIBBON_TOUGH_GREAT] = { 0x8C, 0x5A, 0x4, ribbon_name_tough_great, 0x7B },
-    [RIBBON_TOUGH_ULTRA] = { 0x8D, 0x5B, 0x4, ribbon_name_tough_ultra, 0x7C },
-    [RIBBON_TOUGH_MASTER] = { 0x8E, 0x5C, 0x4, ribbon_name_tough_master, 0x7D },
-    [RIBBON_ABILITY] = { 0x1A, 0x5D, 0x0, ribbon_name_ability, 0x7E },
-    [RIBBON_GREAT_ABILITY] = { 0x1B, 0x5E, 0x0, ribbon_name_great_ability, 0x7F },
-    [RIBBON_DOUBLE_ABILITY] = { 0x1C, 0x5F, 0x0, ribbon_name_double_ability, 0x80 },
-    [RIBBON_MULTI_ABILITY] = { 0x1D, 0x60, 0x0, ribbon_name_multi_ability, 0x81 },
-    [RIBBON_PAIR_ABILITY] = { 0x1E, 0x61, 0x0, ribbon_name_pair_ability, 0x82 },
-    [RIBBON_WORLD_ABILITY] = { 0x1F, 0x62, 0x0, ribbon_name_world_ability, 0x83 },
-    [RIBBON_ALERT] = { 0x20, 0x63, 0x2, ribbon_name_alert, 0x84 },
-    [RIBBON_SHOCK] = { 0x21, 0x64, 0x0, ribbon_name_shock, 0x85 },
-    [RIBBON_DOWNCAST] = { 0x22, 0x65, 0x1, ribbon_name_downcast, 0x86 },
-    [RIBBON_CARELESS] = { 0x23, 0x66, 0x2, ribbon_name_careless, 0x87 },
-    [RIBBON_RELAX] = { 0x24, 0x67, 0x3, ribbon_name_relax, 0x88 },
-    [RIBBON_SNOOZE] = { 0x25, 0x68, 0x0, ribbon_name_snooze, 0x89 },
-    [RIBBON_SMILE] = { 0x26, 0x69, 0x2, ribbon_name_smile, 0x8A },
-    [RIBBON_GORGEOUS] = { 0x27, 0x6A, 0x1, ribbon_name_gorgeous, 0x8B },
-    [RIBBON_ROYAL] = { 0x28, 0x6B, 0x3, ribbon_name_royal, 0x8C },
-    [RIBBON_GORGEOUS_ROYAL] = { 0x29, 0x6C, 0x0, ribbon_name_gorgeous_royal, 0x8D },
-    [RIBBON_FOOTPRINT] = { 0x2A, 0x6D, 0x0, ribbon_name_footprint, 0x8E },
-    [RIBBON_RECORD] = { 0x2B, 0x6E, 0x1, ribbon_name_record, 0x8F },
-    [RIBBON_HISTORY] = { 0x2C, 0x6F, 0x3, ribbon_name_history, 0x90 },
-    [RIBBON_LEGEND] = { 0x2D, 0x70, 0x0, ribbon_name_legend, 0x91 },
-    [RIBBON_RED] = { 0x2E, 0x71, 0x0, ribbon_name_red, 0x8007 },
-    [RIBBON_GREEN] = { 0x2F, 0x72, 0x3, ribbon_name_green, 0x8008 },
-    [RIBBON_BLUE] = { 0x30, 0x73, 0x1, ribbon_name_blue, 0x8009 },
-    [RIBBON_FESTIVAL] = { 0x31, 0x74, 0x1, ribbon_name_festival, 0x800A },
-    [RIBBON_CARNIVAL] = { 0x32, 0x75, 0x0, ribbon_name_carnival, 0x800B },
-    [RIBBON_CLASSIC] = { 0x33, 0x76, 0x1, ribbon_name_classic, 0x800C },
-    [RIBBON_PREMIER] = { 0x34, 0x77, 0x0, ribbon_name_premier, 0x800D },
+    [RIBBON_HOENN_CHAMPION] = {
+        .ribbonID = 98,
+        .spriteID = hoenn_champion_NCGR,
+        .paletteNum = 0,
+        .nameID = ribbon_name_hoenn_champion,
+        .descriptionID = 80,
+    },
+    [RIBBON_HOENN_COOL] = {
+        .ribbonID = 78,
+        .spriteID = hoenn_contest_NCGR,
+        .paletteNum = 0,
+        .nameID = ribbon_name_hoenn_cool,
+        .descriptionID = 81,
```

Comment:
**question**: Are these text bank entries? Do we know where they are?

Reply:
Yep, they're documented now.